sympy/series/formal.py

```diff
@@ -0,0 +1,1107 @@
+    """Represents Formal Power Series of a function.
+
+    No computation is performed. This class should only to be used to represent
+    a series. No checks are performed.
+
+    For computing a series use :func:`fps`.
+
+    See Also
+    ========
+
+    sympy.series.formal.fps
+    sympy.series.formal.compute_fps
+    """
+    def __new__(cls, *args):
+        args = map(sympify, args)
+        return Expr.__new__(cls, *args)
+
+    @property
+    def function(self):
+        return self.args[0]
+
+    @property
+    def x(self):
+        return self.args[1]
+
+    @property
+    def x0(self):
+        return self.args[2]
+
+    @property
+    def ak(self):
+        return self.args[4][0]
+
+    @property
+    def xk(self):
+        return self.args[4][1]
+
+    @property
+    def ind(self):
+        return self.args[4][2]
+
+    @property
+    def mul(self):
+        return self.args[4][3]
+
+    @property
+    def interval(self):
+        return Interval(0, oo)
+
+    @property
+    def start(self):
+        return self.interval.inf
+
+    @property
+    def stop(self):
+        return self.interval.sup
+
+    @property
+    def length(self):
+        return oo
+
+    @property
+    def infinite(self):
+        """Returns an infinite representation of the series"""
+        from sympy.concrete import Sum
+
+        ak, xk = self.ak, self.xk
+        k = ak.variables[0]
+        ind = (self.ind * self.mul).expand()
+        inf_sum = Sum(ak.formula * xk.formula * self.mul,
+                      (k, ak.start, ak.stop))
+
+        return ind + inf_sum
+
+    def _get_pow_x(self, term):
+        """Returns the power of x in a term."""
+        xterm, pow_x = term.as_independent(self.x)[1].as_base_exp()
+        if not xterm.has(self.x):
+            return S.Zero
+        return pow_x
+
+    def polynomial(self, n=6):
+        """Truncated series as polynomial.
+
+        Returns series sexpansion of ``f`` upto order ``O(x**n)``
+        as a polynomial(without ``O`` term).
+        """
+        terms = []
+        for i, t in enumerate(self):
+            xp = self._get_pow_x(t)
+            if xp >= n:
+                break
+            elif xp.is_integer is True and i == n + 1:
+                break
+            elif t is not S.Zero:
+                terms.append(t * self.mul)
+
+        return Add(*terms)
+
+    def truncate(self, n=6):
+        """Truncated series.
+
+        Returns truncated series expansion of f upto
+        order ``O(x**n)``.
+
+        If n is ``None``, returns an infinite iterator.
+        """
+        if n is None:
+            return iter(self)
+
+        x, x0 = self.x, self.x0
+        pt_xk = self.xk.coeff(n)
+        if x0 is S.NegativeInfinity:
+            x0 = S.Infinity
+
+        return self.polynomial(n) + self.mul * Order(pt_xk, (x, x0))
+
+    def _eval_term(self, pt):
+        pt_xk = self.xk.coeff(pt)
+
+        try:
+            pt_ak = self.ak.coeff(pt).simplify()  # TODO: Thoughts?
```

Comment:
I remember discussing this TODO earlier. If the terms `simplify` is being called on are small-ish, then this is probably fine (but not ideal). I've found success in the past with simplifying as an expression is computed being significantly faster than waiting to simplify at the end, so I'm not against this universally. I'll leave it up to your judgement. However, the TODO should be changed to a comment explaining why you simplify here (if you leave it in).